packages/db/src/query/device.ts

```diff
@@ -57,6 +58,80 @@ export async function deleteDevice(id: number, userId: string) {
     });
 }
 
+export async function updatePowerOfDevices(userId: string) {
+    return db.transaction(async (trx) => {
+        const devicesWithPeaks = await trx
+            .select()
+            .from(device)
+            .leftJoin(deviceToPeak, eq(deviceToPeak.deviceId, device.id))
+            .leftJoin(sensorDataSequence, eq(sensorDataSequence.id, deviceToPeak.sensorDataSequenceId))
+            .where(eq(device.userId, userId));
+
+        const devices = Array.from(new Set(devicesWithPeaks.map((device) => device.device.id)));
+        const flattenPeak = devicesWithPeaks
```

Comment:
Hier sollten wir aus Gründen der Verständlichkeit Kommentare einfügen oder die methode in aussagekräftige SubMethoden aufgliedern